cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,506 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/leaderelection"
+	"k8s.io/kubernetes/pkg/client/record"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/flag"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/tpm"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// Flag a node as untrusted
+func invalidateNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+        newTaints := []api.Taint{}
+        untrustedTaint := api.Taint{
+                Key:    "Untrusted",
+                Value:  "true",
+                Effect: api.TaintEffectNoSchedule,
+        }
+        taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+        for _, taint := range taints {
+                if taint.Key == "Untrusted" {
+                        continue
+                }
+                newTaints = append(newTaints, taint)
+        }
+        newTaints = append(newTaints, untrustedTaint)
+        jsonContent, err := json.Marshal(newTaints)
+        node.Annotations[api.TaintsAnnotationKey] = string(jsonContent)
+        return err
+}
+
+func validateNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+        newTaints := []api.Taint{}
+        taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+        for _, taint := range taints {
+                if taint.Key == "Untrusted" {
+                        continue
+                }
+                newTaints = append(newTaints, taint)
+        }
+        jsonContent, err := json.Marshal(newTaints)
+        node.Annotations[api.TaintsAnnotationKey] = string(jsonContent)
+        return err
+}
+
+func isTrusted(node *api.Node) bool {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		return false
+	}
+        for _, taint := range taints {
+                if taint.Key == "Untrusted" {
+			return false
+		}
+	}
+	return true
+}
+
+func getPolicy() (err error) {
```

Comment:
It seems like this function (and a few of the following) could be methods on the TPMManager, rather than relying on a global var. e.g.

`func (m *TPM Manager) getPolicy() error`